src/zjs_spi.h

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) 2017, Intel Corporation.
+
+#ifndef __zjs_spi_h__
+#define __zjs_spi_h__
+
+#include "jerryscript.h"
+
+enum spi_topology {
+    ZJS_TOPOLOGY_FULL_DUPLEX,
+    ZJS_TOPOLOGY_READ,
+    ZJS_TOPOLOGY_WRITE,
+    ZJS_TOPOLOGY_MULTIPLEXED,
+    ZJS_TOPOLOGY_DAISY_CHAIN
+};
+
+enum spi_direction {
+    ZJS_SPI_DIR_READ,
+    ZJS_SPI_DIR_WRITE,
+    ZJS_SPI_DIR_READ_WRITE
+};
+
+typedef struct spi_handle {
+    struct device *spi_device;
+    enum spi_topology topology;
```

Comment:
I see that you don't keep the direction in the handle, that's why i think you could've drop the direction.